pandas/indexes/numeric.py

```diff
@@ -158,14 +162,20 @@ def equals(self, other):
         try:
             return com.array_equivalent(com._values_from_object(self),
                                         com._values_from_object(other))
-        except TypeError:
+        except (TypeError, ValueError):
             # e.g. fails in numpy 1.6 with DatetimeIndex #1681
```

Comment:
remove the comment (as we don't have 1.6), add comment being more descriptive why this should do this